core/helpers.py

```diff
@@ -11,6 +11,11 @@
 )
 
 
+def get_subclass(base_class, subclass_name):
+    subclasses = {cls.__name__: cls for cls in base_class.__subclasses__()}
```

Comment:
I really think this should be a class method of `Observable`.

Moreover, you do not need to parse all the subclasses before returning the good one (waste of time), and you don't need to create a dictionary containing all the subclasses (waste of memory). You can stop when you have found the good one, and you don't need to store the useless classes:
```Python
class Node(YetiDocument):

    [...]

    @classmethod
    def subclass_from_name(cls, subclass_name):
        for subcls in cls.__subclasses__():
            if subcls.__name__ == subclass_name:
                return subcls
```

You would then use it as:
```Python
observable_type = Observable.subclass_from_name(force_type)
```

Reply:
Yup, this looks like the cleanest way to do it. Also the caller should check the return value of this function before doing anything with it (otherwise they will try to instantiate `None`).

Can you please update the docstring to reflect this?

Reply:
@tomchop you're right. I've updated my comment.

Reply:
@p-l- ok much better than my code :) But if I check guess_type function, it raises an exception inside the function. Would not be better to raise also inside subclass_fom_name ?

Reply:
Yep, could be.

Reply:
I think `subclass_from_name` should raise an exception only if the name is invalid (i.e. does not match an existing class name). `guess_type` raises an `ObservableValidationError` when nothing could be guessed, which in this case is dealt with at high-level (so that callers of observable-creating functions can deal with errors)